.github/workflows/browser-tests.yml

```diff
@@ -0,0 +1,114 @@
+              uses: ./.github/actions/setup-job
+
+            - name: Print Firefox version
+              run: firefox --version
+
+            - name: Tachometer the changed packages
+              run: yarn test:changed --browser=firefox
+
+            - name: Create a dummy file to ensure at least one results file exists
+              run: touch tachometer.firefox-ran.txt
+
+            - name: Archive Firefox tachometer results
+              uses: actions/upload-artifact@v4
+              with:
+                  name: tachometer-results-firefox
+                  path: |
+                      tach-results.firefox.*.json
+                      tachometer.firefox-ran.txt
+
+    compare-chrome:
+        name: Compare performance to latest release on Chrome
+
+        # The job will only run if the pull request is from the same repository.
+        # Benchmarks can't run on PRs from forked repos due to comment posting restrictions without a GitHub token.
+        if: github.event.pull_request == null || github.event.pull_request.head.repo.full_name == github.repository
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout main
+              uses: actions/checkout@v4
+              with:
+                  ref: main
+
+            - name: Checkout PR branch
+              uses: actions/checkout@v4
+
+            - name: Setup Job and Install Dependencies
+              uses: ./.github/actions/setup-job
+
+            - name: Tachometer the changed packages
+              run: yarn test:changed
+
+            - name: Create a dummy file to ensure at least one results file exists
+              run: touch tachometer.chrome-ran.txt
+
+            - name: Archive Chrome tachometer results
+              uses: actions/upload-artifact@v4
+              with:
+                  name: tachometer-results-chrome
+                  path: |
+                      tach-results.*.json
+                      tachometer.chrome-ran.txt
+
+    comment-performance:
+        name: Comment tachometer performance results
+        needs: [compare-chrome, compare-firefox]
+
+        # The job will only run if the pull request is from the same repository.
+        # Benchmarks can't run on PRs from forked repos due to comment posting restrictions without a GitHub token.
+        if: github.event.pull_request == null || github.event.pull_request.head.repo.full_name == github.repository
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout main
+              uses: actions/checkout@v4
+              with:
+                  ref: main
+
+            - name: Checkout PR branch
+              uses: actions/checkout@v4
+
+            - name: Setup Job and Install Dependencies
+              uses: ./.github/actions/setup-job
+
+            - name: Retrieve tachometer results archives
+              uses: actions/download-artifact@v4
+              with:
+                  name: tachometer-results-firefox
```

Comment:
I believe you can omit the `name`. If you don't specify it, all artifacts should get downloaded to their respective folders, so you can save the step below.